openslides/agenda/views.py

```diff
@@ -126,33 +126,31 @@ def manage_speaker(self, request, pk=None):
 
         else:
             # request.method == 'DELETE'
-            # Retrieve speaker_id
-            speaker_id = request.data.get('speaker')
+            speaker_ids = request.data.get('speaker')
 
             # Check permissions and other conditions. Get speaker instance.
-            if speaker_id is None:
+            if speaker_ids is None:
                 # Remove oneself
                 queryset = Speaker.objects.filter(
                     item=item, user=self.request.user).exclude(weight=None)
                 try:
-                    # We assume that there aren't multiple entries because this
-                    # is forbidden by the Manager's add method. We assume that
-                    # there is only one speaker instance or none.
                     speaker = queryset.get()
+                    speaker.delete()
```

Comment:
These two lines should be outside the try-except-block because the exception can not occur here.